### DISCUSSION TITLE
Typescript Framer Motion and Chakra components

### DISCUSSION THREAD

**Discussion by ViaxCo** [chakra-ui/chakra-ui#3645] [Q&A] [answered]
From the [docs](https://chakra-ui.com/guides/integrations/with-framer), an example was shown of how to merge props to create a component with motion and chakra when using Typescript. I want to create a motion component using an existing chakra component and the docs don't provide a guide for this use case:

```jsx
type Merge<P, T> = Omit<P, keyof T> & T;
type MotionFlexProps = Merge<HTMLChakraProps<"div">, HTMLMotionProps<"div">>;
const MotionFlex: React.FC<MotionFlexProps> = motion(Flex);

<MotionFlex align="center" justify="center" />
```
"align" and "justify" throw Typescript errors, and I understand that to be because `FlexProps` were not passed to create the component, but I'm not sure how to create it.

**Comment by ViaxCo** [comment, marked as answer]
```diff
type Merge<P, T> = Omit<P, keyof T> & T;
- type MotionFlexProps = Merge<HTMLChakraProps<"div">, HTMLMotionProps<"div">>;
+ type MotionFlexProps = Merge<FlexProps, HTMLMotionProps<"div">>;
const MotionFlex: React.FC<MotionFlexProps> = motion(Flex);
```

Importing the `FlexProps` interface and using it in place of the generic type was able to fix it for me.

**Reply by theoparis** [reply]
This doesn't seem to allow me to autocomplete the chkara flex props though 😕
